### PR TITLE
feat/vitest plugin

### PR DESCRIPTION
**Pull request by adriencaccia**
- chore(moon): bump moon
- docs: add introspection docs
- feat: move getGitDir to core
- feat: move tryIntrospect into plugins and check if required flags are present
- chore: add some benches to plugins
- chore: change ts module
- feat(vitest-plugin): add vitest-plugin
- chore(ci): run on local codspeed
- chore(ci): remove useless node versions
- chore(moon): update deps

### CODE REVIEW

File: packages/vitest-plugin/src/runner.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const gitDir = getGitDir(suite.filepath);
+  if (gitDir === undefined) {
+    throw new Error("Could not find a git repository");
+  }
+  suite.name = path.relative(gitDir, suite.filepath);
+}
+
+class CodSpeedRunner extends NodeBenchmarkRunner {
+  async runSuite(suite: Suite): Promise<void> {
+    logCodSpeed(`PROCESS PID: ${process.pid} in ${__filename}`);
```
Comment (art049) [inline packages/vitest-plugin/src/runner.ts:77]: This could be removed

File: packages/vitest-plugin/src/runner.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (task.meta?.benchmark) benchmarkGroup.push(task as Benchmark);
+    else if (task.type === "suite") benchmarkSuiteGroup.push(task);
+  }
+
+  const currentSuiteName = parentSuiteName
+    ? parentSuiteName + "::" + suite.name
+    : suite.name;
+
+  // run nested suites first, sequentially
+  if (benchmarkSuiteGroup.length) {
```
Comment (art049) [inline packages/vitest-plugin/src/runner.ts:37]: this condition seems useless

File: packages/vitest-plugin/src/runner.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    for (const subSuite of benchmarkSuiteGroup) {
+      await runBenchmarkSuite(subSuite, currentSuiteName);
+    }
+  }
+
+  // return early if there are no benchmarks to run
+  if (benchmarkGroup.length === 0) {
```
Comment (art049) [inline packages/vitest-plugin/src/runner.ts:44]: this one also

File: packages/vitest-plugin/src/runner.ts (hunk 1 above)
Comment (adriencaccia) [reply packages/vitest-plugin/src/runner.ts:77]: Do not remove but put it behind a `debugLog` function defined in `@codspeed/core` that only log when `process.env.CODSPEED_DEBUG === "true"`

**Pull request merged**